### PR TITLE
FINERACT-1932: Fineract modularization - custom modules

### PR DESCRIPTION
**Pull request by marta-jankovics**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-core/src/main/java/org/apache/fineract/portfolio/note/domain/NoteType.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * under the License.
+ */
+package org.apache.fineract.portfolio.note.domain;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.fineract.infrastructure.core.data.EnumOptionData;
+
+public enum NoteType {
```
Comment (galovics) [inline fineract-core/src/main/java/org/apache/fineract/portfolio/note/domain/NoteType.java:26]: Can we revert recreating this file and stick to moving it? Thanks.

File: fineract-core/src/main/java/org/apache/fineract/interoperation/domain/InteropIdentifierType.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ */
+package org.apache.fineract.interoperation.domain;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+@Getter
+public enum InteropIdentifierType {
```
Comment (galovics) [inline fineract-core/src/main/java/org/apache/fineract/interoperation/domain/InteropIdentifierType.java:27]: Can we revert recreating this file and stick to moving it? Thanks.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-core/src/main/java/org/apache/fineract/portfolio/note/domain/NoteType.java (hunk 1 above)
Comment (marta-jankovics) [reply fineract-core/src/main/java/org/apache/fineract/portfolio/note/domain/NoteType.java:26]: Hi Arnold, it was not recreated, it was moved and a bit extended. Git detects this two steps as recreation and I am afraid I cannot do anything about it. Same for InteropIdentifierType. Could you please accept this?

**Review by adamsaghy — APPROVED** [review]
LGTM

File: fineract-core/src/main/java/org/apache/fineract/portfolio/note/domain/NoteType.java (hunk 1 above)
Comment (galovics) [reply fineract-core/src/main/java/org/apache/fineract/portfolio/note/domain/NoteType.java:26]: Not sure why the reorganization caused any changes to be made but whatever. Next time please take extra care on this and do as small changes as possible.

**Review by galovics — APPROVED** [review]

**Pull request merged**